### PR TITLE
Add release notes for 2.16.5

### PR DESCRIPTION
**Pull request by abrookins**
### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,67 @@
...
+- Integrate `CsrfMiddleware` into server — https://github.com/PrefectHQ/prefect/pull/12303
+- Add CSRF support to orchestration client — https://github.com/PrefectHQ/prefect/pull/12314
+- Return 422 when CSRF is disabled and delete expired tokens — https://github.com/PrefectHQ/prefect/pull/12342
+- [pydantic v2] init compat layer — https://github.com/PrefectHQ/prefect/pull/12345
+- Implement CSRF support in the UI — https://github.com/PrefectHQ/prefect/pull/12354
+
+### Documentation
+- Add upstream dependencies guide to docs — https://github.com/PrefectHQ/prefect/pull/12351
+- Documentation updates for event and metric triggers — https://github.com/PrefectHQ/prefect/pull/12366
+- Documentation for the new compound and sequence triggers — https://github.com/PrefectHQ/prefect/pull/12374
```
Comment (serinamarie) [inline RELEASE-NOTES.md:30]:
```suggestion
- Update documentation on event and metric automation triggers — https://github.com/PrefectHQ/prefect/pull/12366
- Add documentation on compound and sequence automation triggers — https://github.com/PrefectHQ/prefect/pull/12374
```

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,67 @@
 # Prefect Release Notes
 
+## Release 2.16.5
+
+### Exciting New Features 🎉
+- Render artifacts and states on the flow run graph by default. — https://github.com/PrefectHQ/prefect/pull/12300
+
+### Enhancements
+- Keep artifacts file in prefect-client — https://github.com/PrefectHQ/prefect/pull/12316
```
Comment (serinamarie) [inline RELEASE-NOTES.md:9]:
```suggestion
### Enhancements
- Render artifacts and states on the flow run graph by default. — https://github.com/PrefectHQ/prefect/pull/12300
- Keep artifacts file in prefect-client — https://github.com/PrefectHQ/prefect/pull/12316
```

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,67 @@
...
+- Add a select all flow runs checkbox to the flow runs page. — https://github.com/PrefectHQ/prefect/pull/12356
+- remove feature flagging around enhanced-deployment-experiment — https://github.com/PrefectHQ/prefect/pull/12360
```
Comment (serinamarie) [inline RELEASE-NOTES.md:11]: @jakekaplan is this worth highlighting or do you know if this was already highlighted before?

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,5 +1,67 @@
...
+
+### Fixes
+- Refactor logic to set `task_key` for background tasks — https://github.com/PrefectHQ/prefect/pull/12337
+- Correct a memory leak with the outbound task run websockets — https://github.com/PrefectHQ/prefect/pull/12346
+
+### Experimental
+- Create `CsrfToken` model and utilities — https://github.com/PrefectHQ/prefect/pull/12289
+- Create `csrf_token` endpoint to generate tokens for clients — https://github.com/PrefectHQ/prefect/pull/12297
+- Integrate `CsrfMiddleware` into server — https://github.com/PrefectHQ/prefect/pull/12303
+- Add CSRF support to orchestration client — https://github.com/PrefectHQ/prefect/pull/12314
```
Comment (serinamarie) [inline RELEASE-NOTES.md:22]: maybe these should be like Prefect Server and Prefect client

Comment (abrookins) [reply RELEASE-NOTES.md:20]: Hmm, yeah. "Prefect Server" technically isn't a thing in 2.0, or it is, depending on who you ask. But maybe I can live in both realities with the right verbiage.

Comment (serinamarie) [reply RELEASE-NOTES.md:20]: ah yeah, true true. I think we say "Prefect server" now in the docs

Comment (serinamarie) [reply RELEASE-NOTES.md:20]: e.g. "a Prefect server"

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,5 +1,67 @@
 # Prefect Release Notes
 
+## Release 2.16.5
+
+### Exciting New Features 🎉
+- Render artifacts and states on the flow run graph by default. — https://github.com/PrefectHQ/prefect/pull/12300
+
+### Enhancements
+- Keep artifacts file in prefect-client — https://github.com/PrefectHQ/prefect/pull/12316
+- Add a select all flow runs checkbox to the flow runs page. — https://github.com/PrefectHQ/prefect/pull/12356
```
[the remark this answers is not in the document]
Comment (collincchoy) [reply RELEASE-NOTES.md:11]: at the very least, we can clump 3 discrete things together. i need to scrape together some PR links... brb.

File: RELEASE-NOTES.md (hunk 3 above)
Comment (jakekaplan) [reply RELEASE-NOTES.md:12]: @serinamarie I think this was highlighted before because we defaulted it to `True` and made it opt-out?

Comment (jakekaplan) [reply RELEASE-NOTES.md:12]: Confirmed with @pleek91 it was highlighted 2 weeks ago

Comment (serinamarie) [reply RELEASE-NOTES.md:12]: niiiice!

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -1,5 +1,52 @@
...
+- Pin Buildkit to 0.12.5 to fix issue with test image build — https://github.com/PrefectHQ/prefect/pull/12343
+- Backporting the Prefect Cloud composite trigger schemas — https://github.com/PrefectHQ/prefect/pull/12378
+
+### Contributors
+* @hainenber
+* @JiginJayaprakash made their first contribution in https://github.com/PrefectHQ/prefect/pull/12307
+* @baisystems made their first contribution in https://github.com/PrefectHQ/prefect/pull/11771
+
+**All changes**: https://github.com/PrefectHQ/prefect/compare/2.16.4...2.16.5
 ## Release 2.16.4
```
Comment (serinamarie) [inline RELEASE-NOTES.md:50]:
```suggestion
**All changes**: https://github.com/PrefectHQ/prefect/compare/2.16.4...2.16.5

## Release 2.16.4
```

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -1,5 +1,52 @@
...
+- Implement CSRF support in the UI — https://github.com/PrefectHQ/prefect/pull/12354
+
+### Documentation
+- Add upstream dependencies guide to docs — https://github.com/PrefectHQ/prefect/pull/12351
+- Update documentation on event and metric automation triggers — https://github.com/PrefectHQ/prefect/pull/12366
+- Add documentation on compound and sequence automation triggers — https://github.com/PrefectHQ/prefect/pull/12374
+- Add CSRF settings to common settings section in docs — https://github.com/PrefectHQ/prefect/pull/12376 
+
+### Uncategorized
+- Pin Buildkit to 0.12.5 to fix issue with test image build — https://github.com/PrefectHQ/prefect/pull/12343
```
Comment (serinamarie) [inline RELEASE-NOTES.md:41]:
```suggestion
- Pin `BuildKit` to 0.12.5 to fix issue with test image build — https://github.com/PrefectHQ/prefect/pull/12343
```

File: RELEASE-NOTES.md [hunk 8]
```diff
@@ -1,5 +1,52 @@
...
+
+### Enhancements
+- Render artifacts and states on the flow run graph by default. — https://github.com/PrefectHQ/prefect/pull/12300
+- Keep artifacts file in prefect-client — https://github.com/PrefectHQ/prefect/pull/12316
+- Add a select all flow runs checkbox to the flow runs page. — https://github.com/PrefectHQ/prefect/pull/12356
+- remove feature flagging around enhanced-deployment-experiment — https://github.com/PrefectHQ/prefect/pull/12360
+- Feature : #11773 UI: Add checkboxes for runs for an individual flow to allow multi-selection/-deletion — https://github.com/PrefectHQ/prefect/pull/12285
+- Retry 500s to make flow and task runs more resilient — https://github.com/PrefectHQ/prefect/pull/12084
+- Add a capability to verify ssl certificates in Prefect CLI — https://github.com/PrefectHQ/prefect/pull/11771
+- Add prefect task-runs to cli — https://github.com/PrefectHQ/prefect/pull/12307
```
Comment (serinamarie) [inline RELEASE-NOTES.md:16]:
```suggestion
- Add `prefect task-run` command group to CLI — https://github.com/PrefectHQ/prefect/pull/12307
```

Comment (serinamarie) [reply RELEASE-NOTES.md:16]: wow, this is a thing

File: RELEASE-NOTES.md [hunk 9]
```diff
@@ -1,5 +1,52 @@
...
+- [background tasks] emit pending event — https://github.com/PrefectHQ/prefect/pull/12352
```
Comment (serinamarie) [inline RELEASE-NOTES.md:17]:
```suggestion
- Correct emit background task state change events — https://github.com/PrefectHQ/prefect/pull/12352
```

File: RELEASE-NOTES.md [hunk 10]
```diff
@@ -1,5 +1,52 @@
...
+- Update `CsrfTokenApi` to retry failed requests due to invalid tokens — https://github.com/PrefectHQ/prefect/pull/12373
+
+### Fixes
+- Refactor logic to set `task_key` for background tasks — https://github.com/PrefectHQ/prefect/pull/12337
+- Correct a memory leak with the outbound task run websockets — https://github.com/PrefectHQ/prefect/pull/12346
+- fix(task/flow): correct type hint for possibly async `@flow` hooks — https://github.com/PrefectHQ/prefect/pull/12231
```
Comment (serinamarie) [inline RELEASE-NOTES.md:23]: maybe more like 'correctly type hint in flow run state change hooks' or something

File: RELEASE-NOTES.md [hunk 11]
```diff
@@ -1,5 +1,51 @@
 # Prefect Release Notes
 
+## Release 2.16.5
+
+### Exciting New Features 🎉
+- The flow run graph UI enhancements for visualizing state changes and artifacts added in 2.16.4 are now enabled by default. — https://github.com/PrefectHQ/prefect/pull/12300
+- It is now easier to bulk select and delete flow runs through the UI. Listings of filterable and selectable flow runs (e.g. on the flow runs, flow, and deployment pages) now include a top-level checkbox for (de)selecting all currently filtered flow runs for bulk deletion. - https://github.com/PrefectHQ/prefect/pull/12356 https://github.com/PrefectHQ/prefect-ui-library/pull/2227 & https://github.com/PrefectHQ/prefect/pull/12285
```
Comment (serinamarie) [inline RELEASE-NOTES.md:7]:
```suggestion
### Multi-select deletion of flow runs
It is now easier to bulk select and delete flow runs through the UI. Listings of filterable and selectable flow runs (e.g. on the flow runs, flow, and deployment pages) now include a top-level checkbox for (de)selecting all currently filtered flow runs for bulk deletion. 

See the following pull request for implementation details:
- https://github.com/PrefectHQ/prefect/pull/12356 
- https://github.com/PrefectHQ/prefect-ui-library/pull/2227
- https://github.com/PrefectHQ/prefect/pull/12285

## Visualize state changes and artifacts in the UI
Additionally, the flow run graph UI enhancements for visualizing state changes and artifacts added in 2.16.4 are now enabled by default. See [the release notes in 2.16.14 for more details](https://github.com/PrefectHQ/prefect/blob/main/RELEASE-NOTES.md#release-2164)!

```

File: RELEASE-NOTES.md [hunk 12]
```diff
@@ -1,5 +1,59 @@
 # Prefect Release Notes
 
+## Release 2.16.5
+
+### Multi-select deletion of flow runs
+It is now easier to bulk select and delete flow runs through the UI. Listings of filterable and selectable flow runs (e.g. on the flow runs, flow, and deployment pages) now include a top-level checkbox for (de)selecting all currently filtered flow runs for bulk deletion. 
+
```
Comment (serinamarie) [inline RELEASE-NOTES.md:7]: @collincchoy would you mind including an OSS UI screenshot here?

File: RELEASE-NOTES.md [hunk 13]
```diff
@@ -1,5 +1,59 @@
...
+See the following pull request for implementation details:
+- https://github.com/PrefectHQ/prefect/pull/12356 
+- https://github.com/PrefectHQ/prefect-ui-library/pull/2227
+- https://github.com/PrefectHQ/prefect/pull/12285
+
+## Visualize state changes and artifacts in the UI
+Additionally, the flow run graph UI enhancements for visualizing state changes and artifacts added in 2.16.4 are now enabled by default. See [the release notes in 2.16.14 for more details](https://github.com/PrefectHQ/prefect/blob/main/RELEASE-NOTES.md#release-2164)!
+
```
Comment (serinamarie) [inline RELEASE-NOTES.md:15]:
```suggestion
```

File: RELEASE-NOTES.md [hunk 14]
```diff
@@ -1,5 +1,60 @@
...
+It is now easier to bulk select and delete flow runs through the UI. Listings of filterable and selectable flow runs (e.g. on the flow runs, flow, and deployment pages) now include a top-level checkbox for (de)selecting all currently filtered flow runs for bulk deletion.
+
+![image](https://github.com/PrefectHQ/prefect/assets/42048900/2431caf4-c1be-4afd-bcff-3c24fa94dc64)
+
+See the following pull request for implementation details:
+- https://github.com/PrefectHQ/prefect/pull/12356 
+- https://github.com/PrefectHQ/prefect-ui-library/pull/2227
+- https://github.com/PrefectHQ/prefect/pull/12285
+
+## Visualize state changes and artifacts in the UI
```
Comment (serinamarie) [inline RELEASE-NOTES.md:15]:
```suggestion
### Visualize state changes and artifacts in the UI
```

File: RELEASE-NOTES.md [hunk 15]
```diff
@@ -1,5 +1,60 @@
...
+- Correct a memory leak with the outbound task run websockets — https://github.com/PrefectHQ/prefect/pull/12346
+- fix(task/flow): correct type hint for possibly async `@flow` hooks — https://github.com/PrefectHQ/prefect/pull/12231
+
+### Experimental
+- Create `CsrfToken` model and utilities — https://github.com/PrefectHQ/prefect/pull/12289
+- Create `csrf_token` endpoint to generate tokens for clients — https://github.com/PrefectHQ/prefect/pull/12297
+- Integrate `CsrfMiddleware` into API server — https://github.com/PrefectHQ/prefect/pull/12303
+- Add CSRF support to client — https://github.com/PrefectHQ/prefect/pull/12314
+- Return 422 when CSRF is disabled and delete expired tokens — https://github.com/PrefectHQ/prefect/pull/12342
+- Add `model_dump` definition for Pydantic v2 compatibility layer — https://github.com/PrefectHQ/prefect/pull/12345
```
Comment (serinamarie) [inline RELEASE-NOTES.md:39]:
```suggestion
- Add `model_dump` definition for Pydantic v2 compatibility layer — https://github.com/PrefectHQ/prefect/pull/12345
- Add experimental `model_json_schema` definition for Pydantic V2 compatibility - https://github.com/PrefectHQ/prefect/pull/12362
```

**Review by collincchoy — APPROVED** [review]
gonna approve to unblock

my stuff lgtm

**Pull request merged**